ocsmesh/raster.py

```diff
@@ -162,6 +163,31 @@ def __iter__(self, chunk_size=None, overlap=None):
         for window in self.iter_windows(chunk_size, overlap):
             yield window, self.get_window_bounds(window)
 
+    @contextmanager
+    def modifying_raster(self, use_src_meta=True, **kwargs):
+        no_except = False
+        try:
+            # pylint: disable=R1732
+            tmpfile = tempfile.NamedTemporaryFile(prefix=tmpdir)
+
+            new_meta = kwargs
+            # Flag to workaround cases where "src" is NOT set yet
+            if use_src_meta:
+                new_meta = self.src.meta.copy()
+                new_meta.update(**kwargs)
+            with rasterio.open(tmpfile.name, 'w', **new_meta) as dst:
+                yield dst
+
+            no_except = True
+
+        finally:
+            if no_except:
+                # So that tmpfile is NOT destroyed when it locally
+                # goes out of scope
+                self._tmpfile = tmpfile
+
+
+
```

Comment:
This helper context manager makes the code overwriting the `tmpfile` cleaner. It also prevents scattering Pylint's R1732 all over raster code.